vic/drivers/cesm/src/cesm_put_data.c

```diff
@@ -147,9 +148,14 @@ vic_cesm_put_data()
 
                 // albedo: direct , visible
                 // CESM units: unitless
-                albedo = AreaFactor *
-                         (atmos->shortwave[NR] /
-                          (atmos->shortwave[NR] - energy.NetShortAtmos));
+                if (atmos->shortwave[NR] > 0.) {
+                    albedo = AreaFactor *
+                             (atmos->shortwave[NR] /
+                              (atmos->shortwave[NR] - energy.NetShortAtmos));
+                }
+                else {
+                    albedo = 0.;
```

Comment:
You should be able to return any albedo where it's dark.  The tricky part is deciding where it's dark and making sure different models on different grids agree on that, especially right along the terminator.  We have had problems with that before.  One interesting test would be to run two cases with two different "dark" albedo values.  You should get identical answers.  Having said that, zeros on the dark side should be fine as long as the models are coupling correctly.  If they aren't, shortwave sent on the "dark side" will be ignore in the land while the atm will think all the radiation sent was absorbed.  It might be better to use 1 because then no shortwave will be lost.  But that also can have problems in cases where you interpolate albedo between grids.  This right answer probably depends on models and grids in use, how the terminator is computed and what different model do with that information.